File: Platforms/Common/Reign.Video.D3D11/ViewPort.cs
using Reign.Core;
using Reign_Video_D3D11_Component;

namespace Reign.Video.D3D11
{
	public class ViewPort : IViewPort
	{
		#region Properties
		private Video video;
		private ViewPortCom com;
		#endregion

		#region Constructors
		public ViewPort(IDisposableResource parent, Point2 position, Size2 size)
		: base(position, size)
		{
			init(parent);
		}

		private void init(IDisposableResource parent)
		{
			video = parent.FindParentOrSelfWithException<Video>();
			com = new ViewPortCom(this.video.com);
		}
		#endregion

		#region Methods
		public override void Apply()
		{
			com.Apply(Position.X, video.BackBufferSize.Height - Size.Height - Position.Y, Size.Width, Size.Height);
		}

		public override void Apply(IRenderTarget renderTarget)
		{
			com.Apply(Position.X, renderTarget.Size.Height - Size.Height - Position.Y, Size.Width, Size.Height);
		}
		#endregion
	}
}

File: Platforms/Common/Reign_Video_D3D11_Component/ViewPort/ViewPort.cpp
#include "pch.h"
#include "ViewPort.h"

namespace Reign_Video_D3D11_Component
{
	#pragma region Constructors
	ViewPortCom::ViewPortCom(VideoCom^ video)
	{
		this->video = video;
	}
	#pragma endregion

	#pragma region Methods
	void ViewPortCom::Apply(int x, int y, int width, int height)
	{
		D3D11_VIEWPORT viewPort =
		{
		    x, y,
		    width, height,
		    0.0f, 1.0f
		};

		video->deviceContext->RSSetViewports(1, &viewPort);
	}
	#pragma endregion
}